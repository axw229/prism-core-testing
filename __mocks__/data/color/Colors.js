// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
