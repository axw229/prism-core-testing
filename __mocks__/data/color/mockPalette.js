const mockPalette = {
  'colors': [
    {
      'colorNumber': '6765',
      'coordinatingColors': {
        'coord2ColorId': '11194',
        'coord1ColorId': '2870',
        'whiteColorId': '2162'
      },
      'description': [
        'Fairly Colorful',
        'Moderately Colorful',
        'Pale'
      ],
      'id': '2452',
      'isExterior': true,
      'isInterior': true,
      'name': 'Spa',
      'lrv': 64.476,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Blue'
      ],
      'brandKey': 'SW',
      'red': 167,
      'green': 220,
      'blue': 220,
      'hue': 0.5,
      'saturation': 0.43089430894308944,
      'lightness': 0.7588235294117647,
      'hex': '#a7dcdc',
      'isDark': false,
      'storeStripLocator': '164-C2',
      'similarColors': [
        '2634',
        '2630',
        '11225',
        '2472',
        '2445',
        '2622',
        '2459',
        '1606',
        '11226',
        '2451'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '6860',
      'coordinatingColors': {
        'coord2ColorId': '11339',
        'coord1ColorId': '11364',
        'whiteColorId': '2681'
      },
      'description': [
        'Soft',
        'Fairly Bright'
      ],
      'id': '2547',
      'isExterior': true,
      'isInterior': true,
      'name': 'Eros Pink',
      'lrv': 18.874,
      'brandedCollectionNames': [
        '2020 Play',
        '2019 Naturalist'
      ],
      'colorFamilyNames': [
        'Red'
      ],
      'brandKey': 'SW',
      'red': 200,
      'green': 79,
      'blue': 104,
      'hue': 0.9655647382920111,
      'saturation': 0.5238095238095238,
      'lightness': 0.5470588235294118,
      'hex': '#c84f68',
      'isDark': false,
      'storeStripLocator': '105-C5',
      'similarColors': [
        '2273',
        '2266',
        '2542',
        '2546',
        '1998',
        '2280',
        '2530',
        '2287',
        '2279',
        '30193'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9018',
      'coordinatingColors': {
        'coord2ColorId': '11218',
        'coord1ColorId': '11200',
        'whiteColorId': '2366'
      },
      'description': [
        'Intense',
        'Pale'
      ],
      'id': '11195',
      'isExterior': false,
      'isInterior': true,
      'name': 'Honey Bees',
      'lrv': 70.14,
      'brandedCollectionNames': [
        '2018 Unity'
      ],
      'colorFamilyNames': [
        'Yellow'
      ],
      'brandKey': 'SW',
      'red': 251,
      'green': 214,
      'blue': 130,
      'hue': 0.11570247933884299,
      'saturation': 0.937984496124031,
      'lightness': 0.7470588235294118,
      'hex': '#fbd682',
      'isDark': false,
      'storeStripLocator': '134-C4',
      'similarColors': [
        '2382',
        '11199',
        '11196',
        '2588',
        '2375',
        '11198',
        '11194',
        '11191',
        '3146',
        '2361'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9007',
      'coordinatingColors': {
        'coord2ColorId': '1666',
        'coord1ColorId': '2686',
        'whiteColorId': '2331'
      },
      'description': [
        'Rich',
        'Vibrant',
        'Pale'
      ],
      'id': '11184',
      'isExterior': false,
      'isInterior': true,
      'name': 'Melón Meloso',
      'lrv': 55.314,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Orange'
      ],
      'brandKey': 'SW',
      'red': 242,
      'green': 184,
      'blue': 140,
      'hue': 0.0718954248366013,
      'saturation': 0.7968749999999999,
      'lightness': 0.7490196078431373,
      'hex': '#f2b88c',
      'isDark': false,
      'storeStripLocator': '121-C3',
      'similarColors': [
        '2334',
        '2983',
        '2327',
        '11188',
        '2326',
        '2045',
        '2038',
        '3012',
        '2341',
        '2031'
      ],
      'ignore': false,
      'archived': false
    }
  ],
  'activeColor': {
    'colorNumber': '9007',
    'coordinatingColors': {
      'coord2ColorId': '1666',
      'coord1ColorId': '2686',
      'whiteColorId': '2331'
    },
    'description': [
      'Rich',
      'Vibrant',
      'Pale'
    ],
    'id': '11184',
    'isExterior': false,
    'isInterior': true,
    'name': 'Melón Meloso',
    'lrv': 55.314,
    'brandedCollectionNames': [],
    'colorFamilyNames': [
      'Orange'
    ],
    'brandKey': 'SW',
    'red': 242,
    'green': 184,
    'blue': 140,
    'hue': 0.0718954248366013,
    'saturation': 0.7968749999999999,
    'lightness': 0.7490196078431373,
    'hex': '#f2b88c',
    'isDark': false,
    'storeStripLocator': '121-C3',
    'similarColors': [
      '2334',
      '2983',
      '2327',
      '11188',
      '2326',
      '2045',
      '2038',
      '3012',
      '2341',
      '2031'
    ],
    'ignore': false,
    'archived': false
  },
  'toggleCompareColor': false,
  'temporaryActiveColor': null
}

export default mockPalette
