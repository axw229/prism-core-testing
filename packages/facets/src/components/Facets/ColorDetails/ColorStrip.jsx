// @flow
import React, { useContext } from 'react'
import { useSelector } from 'react-redux'
import filter from 'lodash/filter'
import split from 'lodash/split'
import type { Color, ColorMap } from 'src/shared/types/Colors.js.flow'
import ConfigurationContext, { type ConfigurationContextType } from 'src/contexts/ConfigurationContext/ConfigurationContext'
import * as GA from 'src/analytics/GoogleAnalytics'
import { GA_TRACKER_NAME_BRAND } from 'src/constants/globals'

type Props = { color: Color, onColorChanged: Color => void }
export function ColorStrip ({ color, onColorChanged }: Props) {
  const { brandId }: ConfigurationContextType = useContext(ConfigurationContext)
  const stripLocation = split(color.storeStripLocator, '-')[0]
  const colorMap: ColorMap = useSelector(store => store.colors.items.colorMap)

  // TODO: Perhaps we need to find a better resolution to the data so we aren't having to do this bright test in multiple places
  const stripColors = stripLocation ? filter(colorMap, c => (!c.ignore && split(c.storeStripLocator, '-')[0] === stripLocation)) : []

  return (stripColors.length > 0 &&
    <ul className={`color-info__strip`}>
      <li className={`color-info__strip-location`}>
        <span className={`color-info__strip-location-name`}>{stripLocation}</span>
      </li>
      {stripColors.map((stripColor, i) => {
        const active = color.id === stripColor.id
        return (
          <li key={i} className='color-info__strip-color' style={{ backgroundColor: stripColor.hex }}>
            <button
              className={`color-info__strip-color-info${active ? ` color-info__strip-color-info--active` : ''}`}
              onClick={() => {
                onColorChanged(stripColor)
                GA.event({ category: 'Color Detail / Swatch Chip List', action: 'View Swatch Chip Color', label: stripColor.name }, GA_TRACKER_NAME_BRAND[brandId])
              }}
            >
              <span className={`color-info__strip-color-name${stripColor.isDark ? ` color-info__strip-color-name--dark-color` : ''}${!active ? ' visually-hidden' : ''}`}>
                {stripColor.name}
              </span>
            </button>
          </li>
        )
      })}
    </ul>
  )
}

export default ColorStrip
