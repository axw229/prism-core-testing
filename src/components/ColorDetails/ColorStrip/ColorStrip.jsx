// @flow
import React, { PureComponent } from 'react'
import { connect } from 'react-redux'
import { filter, split } from 'lodash'

import { paintAllSceneSurfaces } from '../../../actions/scenes'

type Props = {
  colors: Array,
  color: Object,
  paintAllSceneSurfaces: Function
}

class ColorStrip extends PureComponent<Props> {
  activeColor: Color

  constructor (props) {
    super(props)

    this.activeColor = props.color
  }

  render () {
    const stripColors = this.colorStripColors()

    return (
      <React.Fragment>
        <strong>Color Strip</strong>
        <hr />
        <ul>
          {stripColors.map(color => {
            return (
              <li key={color.id}>
                <button onClick={() => this.selectColor(color)}>
                  {color.name}{(color.id === this.activeColor.id) ? ' - ACTIVE' : ''}
                </button>
              </li>
            )
          })}
        </ul>
        <hr />
      </React.Fragment>
    )
  }

  colorStripColors () {
    const { colors, color } = this.props
    const stripLocation = split(color.storeStripLocator, '-')[0]

    // grab all colors with the locationId
    return filter(colors, c => split(c.storeStripLocator, '-')[0] === stripLocation)
  }

  selectColor (color) {
    // track the active color within the color strip
    this.activeColor = color

    // paint all the scenes & secen surfaces
    this.props.paintAllSceneSurfaces(color)
  }
}

const mapDispatchToProps = (dispatch: Function) => {
  return {
    paintAllSceneSurfaces: (color) => {
      dispatch(paintAllSceneSurfaces(color))
    }
  }
}

export default connect(null, mapDispatchToProps)(ColorStrip)
