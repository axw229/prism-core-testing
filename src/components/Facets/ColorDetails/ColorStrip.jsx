// @flow
import React from 'react'
import { useSelector } from 'react-redux'
import filter from 'lodash/filter'
import split from 'lodash/split'
import type { Color, ColorMap } from 'src/shared/types/Colors.js.flow'
import * as GA from 'src/analytics/GoogleAnalytics'

type Props = { color: Color, onColorChanged: Color => void }
export function ColorStrip ({ color, onColorChanged }: Props) {
  const stripLocation = split(color.storeStripLocator, '-')[0]
  const colorMap: ColorMap = useSelector(store => store.colors.items.colorMap)

  // TODO: Perhaps we need to find a better resolution to the data so we aren't having to do this bright test in multiple places
  const stripColors = stripLocation ? filter(colorMap, c => (!c.ignore && split(c.storeStripLocator, '-')[0] === stripLocation)) : []

  return (stripColors.length > 0 &&
    <ul className={`color-info__strip`}>
      <li className={`color-info__strip-location`}>
        <span className={`color-info__strip-location-name`}>{stripLocation}</span>
      </li>
      {stripColors.map(stripColor => {
        const active = color.id === stripColor.id
        return (
          <li className='color-info__strip-color' style={{ backgroundColor: stripColor.hex }}>
            <button
              className={`color-info__strip-color-info${active ? ` color-info__strip-color-info--active` : ''}`}
              onClick={() => {
                onColorChanged(stripColor)
                GA.event({ category: 'Color Detail / Swatch Chip List', action: 'View Swatch Chip Color', label: stripColor.name })
              }}
            >
              <span className={`color-info__strip-color-name${stripColor.isDark ? ` color-info__strip-color-name--dark-color` : ''}${!active ? ' visually-hidden' : ''}`}>
                {stripColor.name}
              </span>
            </button>
          </li>
        )
      })}
    </ul>
  )
}

export default ColorStrip
