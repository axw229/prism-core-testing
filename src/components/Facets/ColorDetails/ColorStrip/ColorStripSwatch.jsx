// @flow
import React, { PureComponent } from 'react'
import type { Color } from '../../../../shared/types/Colors'

type Props = {
  color: Color,
  active: Boolean,
  activateColor: Function,
}

class ColorStripSwatch extends PureComponent<Props> {
  static baseClass = 'color-info'

  constructor (props) {
    super(props)

    this.selectColor = this.selectColor.bind(this)
  }

  render () {
    const { color, active } = this.props

    let BUTTON_CLASSES = [
      `${ColorStripSwatch.baseClass}__strip-color-info`
    ]
    if (active) {
      BUTTON_CLASSES.push(`${ColorStripSwatch.baseClass}__strip-color-info--active`)
    }

    return (
      <li className={`${ColorStripSwatch.baseClass}__strip-color`} style={{ backgroundColor: color.hex }}>
        <button className={BUTTON_CLASSES.join(' ')} onClick={this.selectColor}>
          <span className={`${ColorStripSwatch.baseClass}__strip-color-name ${color.isDark ? `${ColorStripSwatch.baseClass}__strip-color-name--dark-color` : ''}`} >{color.name}</span>
        </button>
      </li>
    )
  }

  selectColor = function selectColor () {
    this.props.activateColor(this.props.color)
  }
}

export default ColorStripSwatch
