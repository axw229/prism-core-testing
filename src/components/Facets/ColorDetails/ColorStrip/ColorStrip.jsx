// @flow
import React, { PureComponent } from 'react'
import { filter, split } from 'lodash'

import ColorStripSwatch from './ColorStripSwatch'

type Props = {
  colors: Array,
  color: Object
}

class ColorStrip extends PureComponent<Props> {
  static baseClass = 'color-info'

  state: State = {
    activeColor: void (0)
  }

  constructor (props) {
    super(props)

    this.state = { activeColor: props.color }

    this.activateColor = this.activateColor.bind(this)
  }

  render () {
    const stripColors = this.colorStripColors()

    if (!stripColors) {
      return null
    }

    return (
      <React.Fragment>
        <ul className={`${ColorStrip.baseClass}__strip`}>
          <li className={`${ColorStrip.baseClass}__strip-location`}>
            <span className={`${ColorStrip.baseClass}__strip-location-name`}>{this.colorStripLocation()}</span>
          </li>
          {stripColors.map(color => {
            return <ColorStripSwatch key={color.id} color={color} activateColor={this.activateColor} active={(this.state.activeColor.id === color.id)} />
          })}
        </ul>
      </React.Fragment>
    )
  }

  colorStripLocation () {
    const { color } = this.props
    const stripLocation = split(color.storeStripLocator, '-')[0]

    return stripLocation
  }

  colorStripColors () {
    const { colors } = this.props
    const stripLocation = this.colorStripLocation()

    if (!stripLocation) {
      return null
    }

    // grab all colors with the locationId
    return filter(colors, c => split(c.storeStripLocator, '-')[0] === stripLocation)
  }

  activateColor (color) {
    this.setState({ activeColor: color })
  }
}

export default ColorStrip
