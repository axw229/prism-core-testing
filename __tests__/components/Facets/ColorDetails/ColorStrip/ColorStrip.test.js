/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { ColorStrip } from 'src/components/Facets/ColorDetails/ColorStrip/ColorStrip'
import * as Colors from '__mocks__/data/color/Colors'
import split from 'lodash/split'

const color = Colors.getColor()
const colors = Colors.getAllColors()
const historyMock = { push: jest.fn() }

const BASE_CLASS = 'color-info'

const getColorStrip = (props) => {
  return shallow(<ColorStrip {...props} />)
}

describe('ColorStrip component with empty props', () => {
  let colorStrip
  beforeEach(() => {
    if (!colorStrip) {
      colorStrip = getColorStrip({ color: [], colors: [], history: historyMock })
    }
  })

  it('ColorStrip is rendering with empty props', () => {
    expect(colorStrip.exists()).toBe(true)
  })
})

describe('ColorStrip component with props', () => {
  let colorStrip
  beforeEach(() => {
    if (!colorStrip) {
      colorStrip = getColorStrip({ colors: colors, color: color, history: historyMock })
    }
  })

  it('ColorStrip is rendering', () => {
    expect(colorStrip.exists()).toBe(true)
  })

  it('ColorStrip is rendering strip location name', () => {
    const stripLocationNameSpan = `span.${BASE_CLASS}__strip-location-name`
    expect(colorStrip.find(stripLocationNameSpan).contains(split(color.storeStripLocator, '-')[0])).toBeTruthy()
  })
})
