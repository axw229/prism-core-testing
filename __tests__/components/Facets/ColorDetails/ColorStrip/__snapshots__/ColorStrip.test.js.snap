// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorStrip component with empty props ColorStrip Snapshot with empty props 1`] = `""`;

exports[`ColorStrip component with props ColorStrip Snapshot with props 1`] = `
<ul
  className="color-info__strip"
>
  <li
    className="color-info__strip-location"
  >
    <span
      className="color-info__strip-location-name"
    >
      102
    </span>
  </li>
  <withRouter(ColorStripSwatch)
    active={true}
    color={
      Object {
        "blue": 219,
        "brandKey": "SW",
        "colorFamilyNames": Array [
          "Red",
        ],
        "colorNumber": "6561",
        "coordinatingColors": Object {
          "coord1ColorId": "2250",
          "coord2ColorId": "1926",
          "whiteColorId": "2681",
        },
        "description": Array [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant",
        ],
        "green": 209,
        "hex": "#ebd1db",
        "hue": 0.935897435897436,
        "id": "2248",
        "isDark": false,
        "isExterior": true,
        "isInterior": true,
        "lightness": 0.8705882352941177,
        "lrv": 68.551,
        "name": "Teaberry",
        "red": 235,
        "saturation": 0.39393939393939387,
        "similarColors": Array [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531",
        ],
        "storeStripLocator": "102-C1",
      }
    }
    key="2248"
  />
  <withRouter(ColorStripSwatch)
    active={false}
    color={
      Object {
        "blue": 207,
        "brandKey": "SW",
        "brandedCollectionNames": Array [
          "Kids' Colors",
        ],
        "colorFamilyNames": Array [
          "Red",
        ],
        "colorNumber": "6562",
        "coordinatingColors": Object {
          "coord1ColorId": "2744",
          "coord2ColorId": "2747",
          "whiteColorId": "2681",
        },
        "description": Array [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant",
        ],
        "green": 192,
        "hex": "#e3c0cf",
        "hue": 0.9285714285714285,
        "id": "2249",
        "isDark": false,
        "isExterior": true,
        "isInterior": true,
        "lightness": 0.8215686274509804,
        "lrv": 58.608,
        "name": "Irresistible",
        "red": 227,
        "saturation": 0.3846153846153844,
        "similarColors": Array [
          "2256",
          "2531",
          "2539",
          "2535",
          "2532",
          "2543",
          "2263",
          "2662",
          "1975",
          "2270",
        ],
        "storeStripLocator": "102-C2",
      }
    }
    key="2249"
  />
  <withRouter(ColorStripSwatch)
    active={false}
    color={
      Object {
        "blue": 173,
        "brandKey": "SW",
        "colorFamilyNames": Array [
          "Red",
        ],
        "colorNumber": "6563",
        "coordinatingColors": Object {
          "coord1ColorId": "2977",
          "coord2ColorId": "2976",
          "whiteColorId": "2681",
        },
        "description": Array [
          "Subdued",
          "Muted",
          "Pale",
        ],
        "green": 154,
        "hex": "#cb9aad",
        "hue": 0.9353741496598639,
        "id": "2250",
        "isDark": false,
        "isExterior": true,
        "isInterior": true,
        "lightness": 0.7,
        "lrv": 38.899,
        "name": "Rosebay",
        "red": 203,
        "saturation": 0.32026143790849665,
        "similarColors": Array [
          "2882",
          "2257",
          "1976",
          "2540",
          "1983",
          "2536",
          "2533",
          "2537",
          "2523",
          "2264",
        ],
        "storeStripLocator": "102-C3",
      }
    }
    key="2250"
  />
  <withRouter(ColorStripSwatch)
    active={false}
    color={
      Object {
        "blue": 147,
        "brandKey": "SW",
        "brandedCollectionNames": Array [
          "Senior Living Color Collection",
        ],
        "colorFamilyNames": Array [
          "Red",
        ],
        "colorNumber": "6564",
        "coordinatingColors": Object {
          "coord1ColorId": "2759",
          "coord2ColorId": "2980",
          "whiteColorId": "2681",
        },
        "description": Array [
          "Diluted",
          "Wan",
          "Bright",
          "Glowing",
        ],
        "green": 126,
        "hex": "#b87e93",
        "hue": 0.9396551724137931,
        "id": "2251",
        "isDark": false,
        "isExterior": true,
        "isInterior": true,
        "lightness": 0.607843137254902,
        "lrv": 27.156,
        "name": "Red Clover",
        "red": 184,
        "saturation": 0.29,
        "similarColors": Array [
          "2258",
          "11178",
          "2538",
          "2265",
          "2541",
          "2534",
          "2537",
          "1976",
          "1977",
          "2664",
        ],
        "storeStripLocator": "102-C4",
      }
    }
    key="2251"
  />
  <withRouter(ColorStripSwatch)
    active={false}
    color={
      Object {
        "blue": 111,
        "brandKey": "SW",
        "brandedCollectionNames": Array [
          "Lifestyle",
        ],
        "colorFamilyNames": Array [
          "Red",
        ],
        "colorNumber": "6565",
        "coordinatingColors": Object {
          "coord1ColorId": "1708",
          "coord2ColorId": "1705",
          "whiteColorId": "2681",
        },
        "description": Array [
          "Diluted",
          "Wan",
          "Moderately Bright",
        ],
        "green": 87,
        "hex": "#92576f",
        "hue": 0.9322033898305085,
        "id": "2252",
        "isDark": true,
        "isExterior": true,
        "isInterior": true,
        "lightness": 0.4568627450980392,
        "lrv": 14.131,
        "name": "Grandeur Plum",
        "red": 146,
        "saturation": 0.25321888412017163,
        "similarColors": Array [
          "1978",
          "2259",
          "1985",
          "1971",
          "2665",
          "1970",
          "2260",
          "2528",
          "1991",
          "1984",
        ],
        "storeStripLocator": "102-C5",
      }
    }
    key="2252"
  />
  <withRouter(ColorStripSwatch)
    active={false}
    color={
      Object {
        "blue": 104,
        "brandKey": "SW",
        "brandedCollectionNames": Array [
          "Lifestyle",
        ],
        "colorFamilyNames": Array [
          "Red",
        ],
        "colorNumber": "6841",
        "coordinatingColors": Object {
          "coord1ColorId": "11364",
          "coord2ColorId": "11212",
          "whiteColorId": "2681",
        },
        "description": Array [
          "Fairly Colorful",
          "Moderately Colorful",
          "Moderately Bright",
        ],
        "green": 61,
        "hex": "#953d68",
        "hue": 0.9185606060606061,
        "id": "2528",
        "isDark": true,
        "isExterior": true,
        "isInterior": true,
        "lightness": 0.411764705882353,
        "lrv": 10.731,
        "name": "Dynamo",
        "red": 149,
        "saturation": 0.419047619047619,
        "similarColors": Array [
          "2529",
          "2260",
          "2253",
          "2665",
          "2530",
          "2252",
          "2267",
          "2259",
          "2527",
          "1978",
        ],
        "storeStripLocator": "102-C6",
      }
    }
    key="2528"
  />
  <withRouter(ColorStripSwatch)
    active={false}
    color={
      Object {
        "blue": 91,
        "brandKey": "SW",
        "colorFamilyNames": Array [
          "Red",
        ],
        "colorNumber": "6842",
        "coordinatingColors": Object {
          "coord1ColorId": "2695",
          "coord2ColorId": "2699",
          "whiteColorId": "2681",
        },
        "description": Array [
          "Fairly Colorful",
          "Moderately Colorful",
          "Dusky",
          "Fairly Dark",
        ],
        "green": 52,
        "hex": "#92345b",
        "hue": 0.9308510638297872,
        "id": "2529",
        "isDark": true,
        "isExterior": true,
        "isInterior": true,
        "lightness": 0.388235294117647,
        "lrv": 9.299,
        "name": "Forward Fuchsia",
        "red": 146,
        "saturation": 0.47474747474747475,
        "similarColors": Array [
          "2528",
          "2260",
          "2253",
          "2267",
          "2530",
          "3071",
          "1979",
          "2274",
          "1999",
          "2549",
        ],
        "storeStripLocator": "102-C7",
      }
    }
    key="2529"
  />
</ul>
`;
